### PR TITLE
Fix the incorrect mapping of the joinMetadata.refColumns fields

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
This PR changes the persist client generation logic to correctly map the `refColumns` fields in `JoinMetadata` to the corresponding SQL table column.

Fixes: https://github.com/ballerina-platform/persist-tools/issues/312

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/syntax/clients/DbClientSyntax.java [hunk 1]
```diff
@@ -365,8 +365,9 @@ private static String getJoinMetaData(Entity entity) {
                         refColumns.append(BalSyntaxConstants.COMMA);
                     }
                     refColumns.append(String.format(BalSyntaxConstants.COLUMN_ARRAY_ENTRY_TEMPLATE,
-                            key.getReference()));
-                    joinColumns.append(String.format(BalSyntaxConstants.COLUMN_ARRAY_ENTRY_TEMPLATE, key.getField()));
+                            BalSyntaxUtils.stripEscapeCharacter(key.getReference())));
+                    joinColumns.append(String.format(BalSyntaxConstants.COLUMN_ARRAY_ENTRY_TEMPLATE,
+                            BalSyntaxUtils.stripEscapeCharacter(key.getField())));
```
Comment (kaneeldias) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/syntax/clients/DbClientSyntax.java:370]: Removes the escape character from the SQL column names

File: persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java [hunk 2]
```diff
@@ -351,53 +351,6 @@ public static void inferRelationDetails(Module entityModule) {
...
-                            assocRelBuilder.setReferences(assockeyColumns.stream().map(Relation.Key::getReference)
-                                    .collect(Collectors.toList()));
-                            assocEntity.getFields().stream().filter(assocfield -> assocfield.getFieldType()
-                                    .equals(entity.getEntityName())).forEach(
-                                    assocField -> {
-                                        assocRelBuilder.setRelationType(
-                                                assocField.isArrayType() ? Relation.RelationType.MANY
-                                                        : Relation.RelationType.ONE);
-                                        assocField.setRelation(assocRelBuilder.build());
-                                    });
```
Comment (kaneeldias) [inline persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java:400]: This section of code is unnecessary as the relevant relations are correctly created in the section above.

Comment (kaneeldias) [reply persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java:400]: This code actually overwrites the correctly inferred relation details with incorrect details.

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/main.bal [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2022 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (kalaiyarasiganeshalingam) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/main.bal:1]:
```suggestion
// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/persist/model.bal [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	OrderItem? orderitem;
+|};
+
+type Order record {|
+    readonly string orderId;
+    string customerId;
+    string createdAt;
+    decimal totalPrice;
+	OrderItem[] orderItems;
+	Payment? payment;
```
Comment (kalaiyarasiganeshalingam) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/persist/model.bal:34]: Shall we format this file.

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_65/main.bal [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2022 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (sahanHe) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_65/main.bal:1]:
```suggestion
// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

Comment (kaneeldias) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_65/main.bal]: Fixed in [01fd48a](https://github.com/ballerina-platform/persist-tools/pull/313/commits/01fd48ae29ea561c5cc0ceb5f600cc0e8072bf45)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/main.bal (hunk 3 above)
Comment (kaneeldias) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/main.bal]: Fixed in [01fd48a](https://github.com/ballerina-platform/persist-tools/pull/313/commits/01fd48ae29ea561c5cc0ceb5f600cc0e8072bf45)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/persist/model.bal (hunk 4 above)
Comment (kaneeldias) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_65/persist/model.bal]: Fixed in [dbb9bdf](https://github.com/ballerina-platform/persist-tools/pull/313/commits/dbb9bdfde426cbb5298aff9b6d0c520990fefb9b)

**Comment by sahanHe** [comment]
LGTM. @kalaiyarasiganeshalingam please approve

**Review by sahanHe — APPROVED** [review]

**Pull request merged**